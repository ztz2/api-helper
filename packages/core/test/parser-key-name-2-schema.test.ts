import ParserKeyName2Schema from '../src/lib/parser/parser-key-name-2-schema';

describe('parser-key-name-2-schema', () => {
  test('.searchValue', async () => {
    const schema = new ParserKeyName2Schema('.searchValue', 'number').parse();
    expect(schema).toMatchSnapshot('.searchValue');
  });
  test('dept.searchValue', async () => {
    const schema = new ParserKeyName2Schema('dept.searchValue', 'number').parse();
    expect(schema).toMatchSnapshot('dept.searchValue');
  });
  test('dept.searchValue.username', async () => {
    const schema = new ParserKeyName2Schema('dept.searchValue.username', 'string').parse();
    expect(schema).toMatchSnapshot('dept.searchValue.username');
  });
  test('roles[0].params', async () => {
    const schema = new ParserKeyName2Schema('roles[0].params', 'boolean').parse();
    expect(schema).toMatchSnapshot('roles[0].params');
  });
  test('roles[0].params.updateBy', async () => {
    const schema = new ParserKeyName2Schema('roles[0].params.updateBy', 'unknown').parse();
    expect(schema).toMatchSnapshot('roles[0].params.updateBy');
  });
  test('[0].params.updateBy[0].time', async () => {
    const schema = new ParserKeyName2Schema('[0].params.updateBy[0].time', 'null').parse();
    expect(schema).toMatchSnapshot('[0].params.updateBy[0].time');
  });
});
