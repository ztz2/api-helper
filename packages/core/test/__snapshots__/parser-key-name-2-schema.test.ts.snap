// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser-key-name-2-schema .searchValue: .searchValue 1`] = `
{
  "targetSchema": NumberSchema {
    "description": "",
    "enum": [],
    "examples": [],
    "format": "",
    "id": "",
    "keyName": "searchValue",
    "label": "",
    "params": [],
    "rawType": "",
    "rules": {
      "exclusiveMaximum": undefined,
      "exclusiveMinimum": undefined,
      "maximum": undefined,
      "minimum": undefined,
      "multipleOf": undefined,
      "required": false,
    },
    "title": "",
    "type": "number",
  },
  "wrapSchema": ObjectSchema {
    "description": "",
    "enum": [],
    "examples": [],
    "format": "",
    "id": "",
    "keyName": "",
    "label": "",
    "params": [
      NumberSchema {
        "description": "",
        "enum": [],
        "examples": [],
        "format": "",
        "id": "",
        "keyName": "searchValue",
        "label": "",
        "params": [],
        "rawType": "",
        "rules": {
          "exclusiveMaximum": undefined,
          "exclusiveMinimum": undefined,
          "maximum": undefined,
          "minimum": undefined,
          "multipleOf": undefined,
          "required": false,
        },
        "title": "",
        "type": "number",
      },
    ],
    "rawType": "object",
    "rules": {
      "required": false,
    },
    "title": "",
    "type": "object",
  },
}
`;

exports[`parser-key-name-2-schema [0].params.updateBy[0].time: [0].params.updateBy[0].time 1`] = `
{
  "targetSchema": NullSchema {
    "description": "",
    "enum": [],
    "examples": [],
    "format": "",
    "id": "",
    "keyName": "time",
    "label": "",
    "params": [],
    "rawType": "null",
    "rules": {
      "required": false,
    },
    "title": "",
    "type": "null",
  },
  "wrapSchema": ArraySchema {
    "description": "",
    "enum": [],
    "examples": [],
    "format": "",
    "id": "",
    "keyName": "",
    "label": "",
    "params": [
      ObjectSchema {
        "description": "",
        "enum": [],
        "examples": [],
        "format": "",
        "id": "",
        "keyName": "",
        "label": "",
        "params": [
          ObjectSchema {
            "description": "",
            "enum": [],
            "examples": [],
            "format": "",
            "id": "",
            "keyName": "params",
            "label": "",
            "params": [
              ArraySchema {
                "description": "",
                "enum": [],
                "examples": [],
                "format": "",
                "id": "",
                "keyName": "updateBy",
                "label": "",
                "params": [
                  ObjectSchema {
                    "description": "",
                    "enum": [],
                    "examples": [],
                    "format": "",
                    "id": "",
                    "keyName": "",
                    "label": "",
                    "params": [
                      NullSchema {
                        "description": "",
                        "enum": [],
                        "examples": [],
                        "format": "",
                        "id": "",
                        "keyName": "time",
                        "label": "",
                        "params": [],
                        "rawType": "null",
                        "rules": {
                          "required": false,
                        },
                        "title": "",
                        "type": "null",
                      },
                    ],
                    "rawType": "object",
                    "rules": {
                      "required": false,
                    },
                    "title": "",
                    "type": "object",
                  },
                ],
                "rawType": "array",
                "rules": {
                  "maxLength": undefined,
                  "minLength": undefined,
                  "required": false,
                  "uniqueItems": undefined,
                },
                "title": "",
                "type": "array",
              },
            ],
            "rawType": "object",
            "rules": {
              "required": false,
            },
            "title": "",
            "type": "object",
          },
        ],
        "rawType": "object",
        "rules": {
          "required": false,
        },
        "title": "",
        "type": "object",
      },
    ],
    "rawType": "array",
    "rules": {
      "maxLength": undefined,
      "minLength": undefined,
      "required": false,
      "uniqueItems": undefined,
    },
    "title": "",
    "type": "array",
  },
}
`;

exports[`parser-key-name-2-schema dept.searchValue.username: dept.searchValue.username 1`] = `
{
  "targetSchema": StringSchema {
    "description": "",
    "enum": [],
    "examples": [],
    "format": "",
    "id": "",
    "keyName": "username",
    "label": "",
    "params": [],
    "rawType": "",
    "rules": {
      "maxLength": undefined,
      "minLength": undefined,
      "pattern": undefined,
      "required": false,
    },
    "title": "",
    "type": "string",
  },
  "wrapSchema": ObjectSchema {
    "description": "",
    "enum": [],
    "examples": [],
    "format": "",
    "id": "",
    "keyName": "dept",
    "label": "",
    "params": [
      ObjectSchema {
        "description": "",
        "enum": [],
        "examples": [],
        "format": "",
        "id": "",
        "keyName": "searchValue",
        "label": "",
        "params": [
          StringSchema {
            "description": "",
            "enum": [],
            "examples": [],
            "format": "",
            "id": "",
            "keyName": "username",
            "label": "",
            "params": [],
            "rawType": "",
            "rules": {
              "maxLength": undefined,
              "minLength": undefined,
              "pattern": undefined,
              "required": false,
            },
            "title": "",
            "type": "string",
          },
        ],
        "rawType": "object",
        "rules": {
          "required": false,
        },
        "title": "",
        "type": "object",
      },
    ],
    "rawType": "object",
    "rules": {
      "required": false,
    },
    "title": "",
    "type": "object",
  },
}
`;

exports[`parser-key-name-2-schema dept.searchValue: dept.searchValue 1`] = `
{
  "targetSchema": NumberSchema {
    "description": "",
    "enum": [],
    "examples": [],
    "format": "",
    "id": "",
    "keyName": "searchValue",
    "label": "",
    "params": [],
    "rawType": "",
    "rules": {
      "exclusiveMaximum": undefined,
      "exclusiveMinimum": undefined,
      "maximum": undefined,
      "minimum": undefined,
      "multipleOf": undefined,
      "required": false,
    },
    "title": "",
    "type": "number",
  },
  "wrapSchema": ObjectSchema {
    "description": "",
    "enum": [],
    "examples": [],
    "format": "",
    "id": "",
    "keyName": "dept",
    "label": "",
    "params": [
      NumberSchema {
        "description": "",
        "enum": [],
        "examples": [],
        "format": "",
        "id": "",
        "keyName": "searchValue",
        "label": "",
        "params": [],
        "rawType": "",
        "rules": {
          "exclusiveMaximum": undefined,
          "exclusiveMinimum": undefined,
          "maximum": undefined,
          "minimum": undefined,
          "multipleOf": undefined,
          "required": false,
        },
        "title": "",
        "type": "number",
      },
    ],
    "rawType": "object",
    "rules": {
      "required": false,
    },
    "title": "",
    "type": "object",
  },
}
`;

exports[`parser-key-name-2-schema roles[0].params.updateBy: roles[0].params.updateBy 1`] = `
{
  "targetSchema": UnknownSchema {
    "description": "",
    "enum": [],
    "examples": [],
    "format": "",
    "id": "",
    "keyName": "updateBy",
    "label": "",
    "params": [],
    "rawType": "unknown",
    "rules": {
      "required": false,
    },
    "title": "",
    "type": "unknown",
  },
  "wrapSchema": ArraySchema {
    "description": "",
    "enum": [],
    "examples": [],
    "format": "",
    "id": "",
    "keyName": "roles",
    "label": "",
    "params": [
      ObjectSchema {
        "description": "",
        "enum": [],
        "examples": [],
        "format": "",
        "id": "",
        "keyName": "",
        "label": "",
        "params": [
          ObjectSchema {
            "description": "",
            "enum": [],
            "examples": [],
            "format": "",
            "id": "",
            "keyName": "params",
            "label": "",
            "params": [
              UnknownSchema {
                "description": "",
                "enum": [],
                "examples": [],
                "format": "",
                "id": "",
                "keyName": "updateBy",
                "label": "",
                "params": [],
                "rawType": "unknown",
                "rules": {
                  "required": false,
                },
                "title": "",
                "type": "unknown",
              },
            ],
            "rawType": "object",
            "rules": {
              "required": false,
            },
            "title": "",
            "type": "object",
          },
        ],
        "rawType": "object",
        "rules": {
          "required": false,
        },
        "title": "",
        "type": "object",
      },
    ],
    "rawType": "array",
    "rules": {
      "maxLength": undefined,
      "minLength": undefined,
      "required": false,
      "uniqueItems": undefined,
    },
    "title": "",
    "type": "array",
  },
}
`;

exports[`parser-key-name-2-schema roles[0].params: roles[0].params 1`] = `
{
  "targetSchema": BooleanSchema {
    "description": "",
    "enum": [],
    "examples": [],
    "format": "",
    "id": "",
    "keyName": "params",
    "label": "",
    "params": [],
    "rawType": "boolean",
    "rules": {
      "required": false,
    },
    "title": "",
    "type": "boolean",
  },
  "wrapSchema": ArraySchema {
    "description": "",
    "enum": [],
    "examples": [],
    "format": "",
    "id": "",
    "keyName": "roles",
    "label": "",
    "params": [
      ObjectSchema {
        "description": "",
        "enum": [],
        "examples": [],
        "format": "",
        "id": "",
        "keyName": "",
        "label": "",
        "params": [
          BooleanSchema {
            "description": "",
            "enum": [],
            "examples": [],
            "format": "",
            "id": "",
            "keyName": "params",
            "label": "",
            "params": [],
            "rawType": "boolean",
            "rules": {
              "required": false,
            },
            "title": "",
            "type": "boolean",
          },
        ],
        "rawType": "object",
        "rules": {
          "required": false,
        },
        "title": "",
        "type": "object",
      },
    ],
    "rawType": "array",
    "rules": {
      "maxLength": undefined,
      "minLength": undefined,
      "required": false,
      "uniqueItems": undefined,
    },
    "title": "",
    "type": "array",
  },
}
`;
